packages/transaction-controller/src/TransactionController.ts

```diff
@@ -4234,4 +4230,100 @@ export class TransactionController extends BaseController<
       newTransactionMeta,
     );
   }
+
+  async #getSimulationData({
```

Comment:
I've an upcoming PR that will better modularise this simulation logic to keep it out of `TransactionController.ts`.